markdownlint-cli2-config-schema.json

```diff
@@ -0,0 +1,83 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"title": "markdownlint-cli2 configuration schema",
+	"type": "object",
+	"properties": {
+		"config": {
+			"$ref": "./node_modules/markdownlint/schema/markdownlint-config-schema.json"
+		},
+		"customRules": {
+			"description": "Module names or paths of custom rules to load and use when linting",
+			"type": "array",
+			"items": {
+				"description": "Module name or path of a custom rule",
+				"type": "string"
+			}
+		},
```

Comment:
Can you please add relevant defaults for all nodes? Here, `"default": null`.

Reply:
I attempted this to be closer to the markdownlint schema, but when using `null` as the default, it would flag both strings and arrays as invalid. Without the `default`specified VS Code would insert an empty string or array, which in my opinion is better DX. See:

<img width="489" alt="Screen Shot 2020-11-28 at 7 43 25 PM" src="https://user-images.githubusercontent.com/831617/100531087-5df44d80-31b7-11eb-9eff-1e015963ddb0.png">

vs.

<img width="235" alt="Screen Shot 2020-11-28 at 7 44 13 PM" src="https://user-images.githubusercontent.com/831617/100531090-664c8880-31b7-11eb-9a73-e157e523d183.png">

I've made it explicit and added empty string and array values as defaults to the top-level schema properties in b1d7282. I tried using empty string and array values on the first array element inside of the `markdownItPlugins` and `outputFormatters` arrays but it looks odd in the editor and technically is not correct. The `minItems` and `minLength` (as described in that [comment response](https://github.com/DavidAnson/vscode-markdownlint/pull/139#discussion_r532148498)) is sufficient to validate these.

Reply:
Would type=array|null address the top issue?

Reply:
Maybe it's good like you have it. The default is actually null, but it would never make sense to put null in a config file instead of omitting the property entirely.